base/loading.jl

```diff
@@ -2312,10 +2312,18 @@ function include_package_for_output(pkg::PkgId, input::String, depot_path::Vecto
     catch ex
         precompilableerror(ex) || rethrow()
         @debug "Aborting `create_expr_cache'" exception=(ErrorException("Declaration of __precompile__(false) not allowed"), catch_backtrace())
-        exit(125) # we define status = 125 means PrecompileableError
+        exit(125) # we define status = 125 means PrecompilableError
     finally
         Core.Compiler.track_newly_inferred.x = false
     end
+
+    # restore some globals in case they were altered in the precompilation workload
+    append!(empty!(Base.DEPOT_PATH), depot_path)
+    append!(empty!(Base.DL_LOAD_PATH), dl_load_path)
```

Comment:
Can you add a comment explaining why it's necessary to restore `Base.DL_LOAD_PATH`?

Reply:
None of these except DEPOT_PATH are necessary to aid the relocation.
I thought it would make sense to reset them too, because they are just set like this before running precompilation workload.

PkgEval showed a few hick ups, perhaps some where caused by these.

Reply:
Hmmm. If the only thing that is necessary for the relocation is the DEPOT_PATH, then let's just do that, and get rid of the other stuff?